static/js/hotkey.js

```diff
@@ -863,6 +863,14 @@ export function process_hotkey(e, hotkey) {
             return true;
     }
 
+    if (
+        // Allow UI only features for spectators which they can perform.
+        page_params.is_spectator &&
+        !["narrow_by_topic", "narrow_by_recipient", "show_lightbox"].includes(event_name)
+    ) {
+        return true;
+    }
```

Comment:
Can't they do `show_sender_info` as well?

Also, what is your thinking on whether we should have a `login_to_access` call here?

Reply:
Oh, yeah!